service/src/main/java/uk/gov/hmcts/reform/fpl/service/DirectionHelperService.java

```diff
@@ -103,10 +210,35 @@ public void persistHiddenDirectionValues(List<Element<Direction>> directionWithV
      */
     public Map<String, List<Element<Direction>>> sortDirectionsByAssignee(List<Element<Direction>> directions) {
         return directions.stream()
-            .filter(x -> x.getValue().getCustom() == null)
             .collect(groupingBy(directionElement -> directionElement.getValue().getAssignee().getValue()));
     }
 
+    /**
+     * Removes directions where custom is set to Yes.
+     *
+     * @param directions any list of directions.
+     * @return a list of directions that are not custom.
+     */
+    public List<Element<Direction>> removeCustomDirections(List<Element<Direction>> directions) {
+        return directions.stream()
+            .filter(element -> !"Yes".equals(element.getValue().getCustom()))
+            .collect(Collectors.toList());
+    }
+
+    /**
+     * Filters a list of directions to return only the directions belonging to specific assignee.
+     *
+     * @param directions a list of directions.
+     * @param assignee   the assignee of the directions to be returned.
+     * @return a list of directions belonging to the assignee.
+     */
+    public List<Element<Direction>> getDirectionsForAssignee(List<Element<Direction>> directions,
+                                                             DirectionAssignee assignee) {
+        return directions.stream()
+            .filter(element -> element.getValue().getAssignee().equals(assignee))
+            .collect(Collectors.toList());
```

Comment:
Nothing major, but it's another instance where we collect just to stream couple of steps later - I think we could be mindful a bit more. What do you think?

Reply:
hmm I want these public methods to be reusable. Hence why I am returning a list so often. It means they can be used anywhere easily, and keeps the methods simple as to what they are doing

Reply:
What's not reusable in Returning Streams? ;) ok I guess some clients need to collect before doing some operations, but if they need to filter or do other operations, they will need to stream. Happy to discuss this at later point though, definitely don't need to make changes right now! 

https://stackoverflow.com/questions/24676877/should-i-return-a-collection-or-a-stream

The last paragraph from the answers sums my thinking pretty nicely as well:

> So I would say that most of the time, Stream is the right answer -- it is more flexible, it doesn't impose usually-unnecessary materialization costs, and can be easily turned into the Collection of your choice if needed. But sometimes, you may have to return a Collection (say, due to strong consistency requirements), or you may want to return Collection because you know how the user will be using it and know this is the most convenient thing for them.